benchmarks/petsc-bps.sh

```diff
@@ -0,0 +1,53 @@
+# Copyright (c) 2017-2018, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory. LLNL-CODE-734707.
+# All Rights reserved. See files LICENSE and NOTICE for details.
+#
+# This file is part of CEED, a collection of benchmarks, miniapps, software
+# libraries and APIs for efficient high-order finite element and spectral
+# element discretizations for exascale applications. For more information and
+# source code availability see http://github.com/ceed.
+#
+# The CEED research is supported by the Exascale Computing Project 17-SC-20-SC,
+# a collaborative effort of two U.S. Department of Energy organizations (Office
+# of Science and the National Nuclear Security Administration) responsible for
+# the planning and preparation of a capable exascale ecosystem, including
+# software, applications, hardware, advanced system engineering and early
+# testbed platforms, in support of the nation's exascale computing imperative.
+
+
+function run_petsc_tests()
```

Comment:
Why this pattern of sourcing and then the function? Also note that you can pass arguments
```
.: . filename [arguments]
    Execute commands from a file in the current shell.
```

Reply:
Honestly, I was repeating the style already there; I was trying not to change too much from what was already there. There are already 9 variables being passed in this fashion from `benchmark.sh`.

Reply:
I dropped this and am using a single script for the PETSc BPs. I think this addresses your concern.